### PR TITLE
WIP: Example usage of a NameServiceProvider to override DNS behaviors.

### PR DESCRIPTION
**Pull request by joshua-rutherford**
NO NOT MERGE: This is only an example solution for #112 and it is not ready without further discussion.  Note that I do not know that this approach works on OpenJDK and it is likely to break in later releases of Java.

Running `mvn clean verify` and looking at the ZooKeeper docker instance that is spun up you can see that the example service is announcing with it's IP address.  If you remove this line, you will see that it announces with its hostname as per DNS: https://github.com/DecipherNow/gm-fabric-jvm/pull/125/files#diff-badf7d07c842086d1446c5fe29e735b9R50.

### CODE REVIEW

**Comment by mikejlong60** [comment]
I am fine with this as long as it will work, as long as we avoid new settings and configuration beyond what we already have. Seems like the original intent was there but had some bugs.

I spent some time looking at the way the GMFNetworkConfigurationResolver in the framework decides whether or not to announce a DNS name or an IP address to ZK.  It relies on these two properties being set:
-com.deciphernow.server.config.ipAddress.enableIpAddressResolution=true
-com.deciphernow.server.config.ipAddress.useNetworkInterfaceName=eth0(or whatever)

What I found is that the GMFNetworkConfigurationResolver class does not go and get the value for these two properties. As a consequence I could never get the application that I was testing against to broadcast anything other than a DNS name.  The bug is that the Twitter Flag function that gets the value(i.e. apply) for the property never gets executed.  And thus the boolean logic that decides what to do ignores the user settings.  Its hard to follow in the program because there is so much indirection.  I had suggested earlier that we refactor that program to use lazy vals instead.

Let me know if you want me to do anything further.

**Comment by joshua-rutherford** [comment]
While this works, after implementing the alternative, I think it is probably more reliable on different JVM implementations.  I'm going to close this in favor of #126.
